From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Sun, 15 Jun 2025 02:20:26 +0800
Subject: [PATCH] Fix crash during parsing unknown command message

Use direct impl for unknown commands message parse to prevent crash
Filter control chars to prevent crash

diff --git a/net/minecraft/commands/Commands.java b/net/minecraft/commands/Commands.java
index 7a4a0df959ca9bf25156e180c6da2b0d46bc4898..41d801c1318ea6639432265d76956cdf2cc0c651 100644
--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -680,7 +680,7 @@ public class Commands {
         net.kyori.adventure.text.Component detailComponent = null;
 
         if (rawMessage.contains("<message>")) {
-            messageComponent = io.papermc.paper.command.brigadier.MessageComponentSerializer.message().deserialize(e.getRawMessage());
+            messageComponent = io.papermc.paper.adventure.PaperAdventure.asAdventure(ComponentUtils.fromMessage(e.getRawMessage())); // Leaf - Fix crash during parsing unknown command message
         }
 
         final String input = e.getInput();
@@ -700,9 +700,9 @@ public class Commands {
                 detail.append(net.kyori.adventure.text.Component.text("..."));
             }
 
-            detail.append(net.kyori.adventure.text.Component.text(input.substring(Math.max(0, min - 10), min)));
+            detail.append(net.kyori.adventure.text.Component.text(net.minecraft.util.StringUtil.filterText(input.substring(Math.max(0, min - 10), min)))); // Leaf - Fix crash during parsing unknown command message
             if (min < input.length()) {
-                net.kyori.adventure.text.Component commandInput = net.kyori.adventure.text.Component.text(input.substring(min))
+                net.kyori.adventure.text.Component commandInput = net.kyori.adventure.text.Component.text(net.minecraft.util.StringUtil.filterText(input.substring(min))) // Leaf - Fix crash during parsing unknown command message
                     .color(net.kyori.adventure.text.format.NamedTextColor.RED)
                     .decorate(net.kyori.adventure.text.format.TextDecoration.UNDERLINED);
 
@@ -723,7 +723,7 @@ public class Commands {
     private static net.kyori.adventure.text.Component getVanillaUnknownCommandMessage(
         final net.kyori.adventure.text.TextComponent.Builder builder, final CommandSyntaxException e, final String commandString
     ) {
-        builder.color(net.kyori.adventure.text.format.NamedTextColor.RED).append(io.papermc.paper.command.brigadier.MessageComponentSerializer.message().deserialize(e.getRawMessage()));
+        builder.color(net.kyori.adventure.text.format.NamedTextColor.RED).append(io.papermc.paper.adventure.PaperAdventure.asAdventure(ComponentUtils.fromMessage(e.getRawMessage()))); // Leaf - Fix crash during parsing unknown command message
 
         if (e.getInput() != null && e.getCursor() >= 0) {
             int cursor = Math.min(e.getInput().length(), e.getCursor());
@@ -734,9 +734,9 @@ public class Commands {
                 context.append(CommonComponents.ELLIPSIS);
             }
 
-            context.append(e.getInput().substring(Math.max(0, cursor - 10), cursor));
+            context.append(net.minecraft.util.StringUtil.filterText(e.getInput().substring(Math.max(0, cursor - 10), cursor))); // Leaf - Fix crash during parsing unknown command message
             if (cursor < e.getInput().length()) {
-                Component remaining = Component.literal(e.getInput().substring(cursor)).withStyle(ChatFormatting.RED, ChatFormatting.UNDERLINE);
+                Component remaining = Component.literal(net.minecraft.util.StringUtil.filterText(e.getInput().substring(cursor))).withStyle(ChatFormatting.RED, ChatFormatting.UNDERLINE); // Leaf - Fix crash during parsing unknown command message
                 context.append(remaining);
             }
 
